### PR TITLE
chore(authn): add MongoDB backend tests

### PR DESCRIPTION
**Pull request by savonarola**

### CODE REVIEW

File: apps/emqx_connector/src/emqx_connector_mongo.erl [hunk 1]
```diff
@@ -178,12 +178,13 @@ health_check(PoolName) ->
 
 %% ===================================================================
 connect(Opts) ->
-    Type = proplists:get_value(mongo_type, Opts, single),
+    Type = proplists:get_value(type, Opts, single),
```
Comment (savonarola) [inline apps/emqx_connector/src/emqx_connector_mongo.erl:181]: @tigercl pls have a look at this change, things seem not be working without it 🤔

Comment (zmstone) [reply apps/emqx_connector/src/emqx_connector_mongo.erl:181]: related change: https://github.com/emqx/emqx/pull/6277/files

**Review by zmstone — COMMENTED** [review]
LGTM. but @tigercl should have a look.

File: apps/emqx_authn/test/emqx_authn_mongo_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+              {null,        false},
+              {false,       false},
+              {0,           false},
+
+              {<<"1">>,     true},
+              {<<"val">>,   true},
+              {1,           true},
+              {123,         true},
+              {true,        true}
+             ],
```
Comment (thalesmg) [inline apps/emqx_authn/test/emqx_authn_mongo_SUITE.erl:208]: Maybe test the new `undefined` case?

Comment (savonarola) [reply apps/emqx_authn/test/emqx_authn_mongo_SUITE.erl:208]: It is caught by this test, we get undefined when insert null atom :)

File: apps/emqx_connector/src/emqx_connector_mongo.erl (hunk 1 above)
Comment (savonarola) [reply apps/emqx_connector/src/emqx_connector_mongo.erl:181]: Yes, I also made them, but after rebase only this one left.

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_connector/src/emqx_connector_mongo.erl (hunk 1 above)
Comment (tigercl) [reply apps/emqx_connector/src/emqx_connector_mongo.erl:181]: `mongo_type` should be used here, using `type` is a bug I introduced. I have fixed it in the PR mentioned by zmstone.

Comment (savonarola) [reply apps/emqx_connector/src/emqx_connector_mongo.erl:181]: @tigercl 

🤔 Here type is still set to `Opts`: https://github.com/emqx/emqx/pull/6277/files#diff-331b796db2f9509a1bf5d10eb34ec3c984fafd2c3f744466cfafde1bbc2fc945R121
And then these opts come to `connect/1` fun.

**Review by k32 — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

File: apps/emqx_connector/src/emqx_connector_mongo.erl (hunk 1 above)
Comment (tigercl) [reply apps/emqx_connector/src/emqx_connector_mongo.erl]: @savonarola `mongo_type` is for the user layer, and `type` is for the driver layer.

**Pull request merged**